test/AutoRest.Shared.Tests/WriterExtensionTests.cs

```diff
@@ -6,6 +6,7 @@
 using System.IO;
 using System.Text.Json;
 using NUnit.Framework;
+using JsonObject = System.Collections.Generic.Dictionary<string, object>;
```

Comment:
Can we add test cases that test the boundaries where we start with '{' but don't end with '}' as well as the other combos we are looking for
Add test cases with various trailing whitespace
Cases with all whitespace

Reply:
added